src/Razor/test/Microsoft.VisualStudio.LanguageServerClient.Razor.Test/RazorLanguageService`IVsLanguageDebugInfoTest.cs

```diff
@@ -196,22 +204,69 @@ public void GetProximityExpressions_CanNotCreateDialog_ReturnsEFail()
                 Mock.Get(proximityExpressionResolver).Setup(r => r.TryResolveProximityExpressionsAsync(It.IsAny<ITextBuffer>(), It.IsAny<int>(), It.IsAny<int>(), It.IsAny<CancellationToken>())).ReturnsAsync(value: null);
             }
 
-            waitDialogFactory ??= new TestWaitDialogFactory();
+            waitDialogFactory ??= new TestIUIThreadOperationExecutor();
             editorAdaptersFactory ??= Mock.Of<IVsEditorAdaptersFactoryService>(service => service.GetDataBuffer(It.IsAny<IVsTextBuffer>()) == new TestTextBuffer(new StringTextSnapshot(Environment.NewLine)), MockBehavior.Strict);
 
-            var languageService = new RazorLanguageService(breakpointResolver, proximityExpressionResolver, waitDialogFactory, editorAdaptersFactory);
+            var languageService = new RazorLanguageService(breakpointResolver, proximityExpressionResolver, waitDialogFactory, editorAdaptersFactory, JoinableTaskFactory);
             return languageService;
         }
 
-        private class TestWaitDialogFactory : WaitDialogFactory
+        private class TestIUIThreadOperationExecutor : IUIThreadOperationExecutor
```

Comment:
Note as long as you have a MEF composition with the editor there is a default implementation that will do all of this for you.

Reply:
After talking offline it sounds like we don't have that, so we'll go forward with this.